crates/holochain/src/conductor/conductor.rs

```diff
@@ -2963,6 +2962,103 @@ mod misc_impls {
             Ok(action_hash)
         }
 
+        /// Get capability grant info for a set of App cells including revoked capabality grants
+        pub async fn capability_grant_info(
+            &self,
+            cell_set: &HashSet<CellId>,
+        ) -> ConductorApiResult<AppCapGrantInfo> {
+            let mut hash_map: HashMap<CellId, Vec<CapGrantInfo>> = HashMap::new();
+            let grant_query: ChainQueryFilter = ChainQueryFilter::new()
+                .include_entries(true)
+                .entry_type(EntryType::CapGrant);
+            let delete_query: ChainQueryFilter = ChainQueryFilter::new()
+                .include_entries(true)
+                .action_type(ActionType::Delete);
+
+            for cell_id in cell_set.iter() {
+                // create a source chain read to query for the cap grant
+                let chain = SourceChainRead::new(
+                    self.get_or_create_authored_db(
+                        cell_id.dna_hash(),
+                        cell_id.agent_pubkey().clone(),
+                    )?
+                    .into(), //authored_db.into(),
+                    self.get_or_create_dht_db(cell_id.dna_hash())?.into(),
+                    self.get_or_create_space(cell_id.dna_hash())?
+                        .dht_query_cache,
+                    self.keystore().clone(),
+                    cell_id.agent_pubkey().clone(),
+                )
+                .await?;
+
+                // query for the cap grant and delete actions (capability revokes)
+                let grant_list = chain.query(grant_query.clone()).await?;
+                if grant_list.is_empty() {
+                    continue;
```

Comment:
This is quite a big loop. It'd be helpful to have comments where the control flow is being affected. For example, something like `// No cap grants for this cell`, would make the code easier to follow